### PR TITLE
Add typed attribute

### PR DESCRIPTION
**Pull request by goganchic**
Based on discussion in https://github.com/apiaryio/mson/issues/66

### CODE REVIEW

File: src/MSONUtility.h [hunk 1]
```diff
@@ -181,6 +181,9 @@ namespace mson {
         else if (attribute == "nullable") {
             typeAttributes |= NullableTypeAttribute;
         }
+        else if (attribute == "typed") {
```
Comment (pksunkara) [inline src/MSONUtility.h]: Please name it `fixed-type`.

File: src/MSON.h [hunk 2]
```diff
@@ -135,7 +135,8 @@ namespace mson {
         FixedTypeAttribute    = (1 << 2),  // The type is fixed
         SampleTypeAttribute   = (1 << 3),  // The type is a sample
         DefaultTypeAttribute  = (1 << 4),  // The type is default
-        NullableTypeAttribute = (1 << 5)   // The type is nullable
+        NullableTypeAttribute = (1 << 5),  // The type is nullable
+        TypedTypeAttribute    = (1 << 6)   // The type is typed
```
Comment (pksunkara) [inline src/MSON.h]: Let's change the comments a bit more since the type attributes are much more now.

File: src/MSONUtility.h (hunk 1 above)
Comment (goganchic) [inline src/MSONUtility.h]: done

File: src/MSON.h (hunk 2 above)
Comment (goganchic) [inline src/MSON.h]: done

File: src/MSON.h [hunk 3]
```diff
@@ -135,7 +135,8 @@ namespace mson {
         FixedTypeAttribute    = (1 << 2),  // The type is fixed
         SampleTypeAttribute   = (1 << 3),  // The type is a sample
         DefaultTypeAttribute  = (1 << 4),  // The type is default
-        NullableTypeAttribute = (1 << 5)   // The type is nullable
+        NullableTypeAttribute = (1 << 5),  // The type is nullable
+        TypedTypeAttribute    = (1 << 6)   // This type works like fixed, but it is unheritable
```
Comment (pksunkara) [inline src/MSON.h]: We can do `FixedTypeTypeAttribute`.

**Comment by pksunkara** [comment]
Only 1 comment, otherwise lgtm.

File: src/MSON.h (hunk 3 above)
Comment (goganchic) [inline src/MSON.h]: done

**Pull request merged**